fizzbuzz.js

```diff
@@ -1,11 +1,80 @@
 // This is our main function
-function fizzbuzz() {
-    console.log("Hello, World!");
+function fizzbuzz(N) {
+    console.log("Let's play FizzBuzz!");
 
-    // Put your code here...
+    function fizz(list) {
+        list.push('Fizz')
+        return list
+    }
 
+    function buzz(list) {
+        list.push('Buzz')
+        return list
+    }
+
+    function bang(list) {
+        list.push('Bang')
+        return list
+    }
+
+    function bong(list) {
+        list = ['Bong']
+        return list
+    }
+
+    function fezz(list) {
+        const first_B = list.findIndex((string) => string.slice(0,1) == 'B')
+        if (first_B == -1) {
+            list.push('Fezz')
+        }
+        else {
+            const start = list.slice(0, first_B)
+            const end = list.slice(first_B)
+            list = start.concat(['Fezz'], end)
```

Comment:
Nice use of arrays to keep track of the strings - I think you can achieve this particular rule using [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) too 😄